src/ds/redblacktree.h

```diff
@@ -77,26 +109,35 @@ namespace snmalloc
     // to treat left, right and root uniformly.
     class ChildRef
     {
-      H* ptr;
+      H ptr;
 
     public:
       ChildRef() : ptr(nullptr) {}
 
-      ChildRef(H& p) : ptr(&p) {}
+      ChildRef(H p) : ptr(p) {}
+
+      ChildRef(const ChildRef& other) = default;
 
       operator K()
       {
         return Rep::get(ptr);
       }
 
-      ChildRef& operator=(const K& t)
+      ChildRef& operator=(const ChildRef& other) = default;
+
+      ChildRef& operator=(const K t)
       {
         // Use representations assigment, so we update the correct bits
         // color and other things way also be stored in the Holder.
         Rep::set(ptr, t);
         return *this;
       }
 
+      /**
+       * Comparison operators.  Note that these are address-comparison, not
```

Comment:
It compares that two Hs have aliased storage (i.e. writing to one will be visible to reads through the other), not that they refer to the same value.  This confused me a lot because I initially assumed comparing two holders would compare the values that they pointed to.